src/Microsoft.VisualStudio.SDK.Analyzers.CodeFixes/VSSDK001DeriveFromAsyncPackageCodeFix.cs

```diff
@@ -30,31 +30,29 @@ public class VSSDK001DeriveFromAsyncPackageCodeFix : CodeFixProvider
         public override ImmutableArray<string> FixableDiagnosticIds => ReusableFixableDiagnosticIds;
 
         /// <inheritdoc />
-        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             Diagnostic diagnostic = context.Diagnostics.First();
 
-            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
-            BaseTypeSyntax baseTypeSyntax = root.FindNode(diagnostic.Location.SourceSpan).FirstAncestorOrSelf<BaseTypeSyntax>();
-            ClassDeclarationSyntax classDeclarationSyntax = baseTypeSyntax.FirstAncestorOrSelf<ClassDeclarationSyntax>();
-
             context.RegisterCodeFix(
                 CodeAction.Create("Convert to async package", ct => this.ConvertToAsyncPackageAsync(context, diagnostic, ct), "only one"),
                 diagnostic);
+
+            return Task.CompletedTask;
         }
 
         /// <inheritdoc />
         public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
 
         private async Task<Document> ConvertToAsyncPackageAsync(CodeFixContext context, Diagnostic diagnostic, CancellationToken cancellationToken)
         {
-            SemanticModel semanticModel = await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            SemanticModel semanticModel = (await context.Document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false))!;
             Compilation? compilation = await context.Document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
             Assumes.NotNull(compilation);
-            SyntaxNode root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            BaseTypeSyntax baseTypeSyntax = root.FindNode(diagnostic.Location.SourceSpan).FirstAncestorOrSelf<BaseTypeSyntax>();
-            ClassDeclarationSyntax classDeclarationSyntax = baseTypeSyntax.FirstAncestorOrSelf<ClassDeclarationSyntax>();
-            MethodDeclarationSyntax initializeMethodSyntax = classDeclarationSyntax.DescendantNodes()
+            SyntaxNode root = (await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false))!;
```

Comment:
I don't think these null forgiveness operators (`!`) are warranted. Would an empty code file have no `root`? Certainly a syntax node may have no `BaseTypeSyntax` as an ancestor, so null might be expected from that. Same for `ClassDeclarationSyntax`. If we closely read the context in which this method is called, we *may* be able to argue that these will never be null, but this is a complex system and we may get it wrong. And even if we're right, the context in which it is called may change over time violating the assumptions we're making here. Can we instead add the appropriate null checks and throw something (more meaningful than NRE) if we cannot complete the work?

Reply:
I added `Assumes.NotNull` instead.
The few `!` left should be warranted by earlier checks.